master/getting-started/openstack/upgrade/upgrade.md

```diff
@@ -0,0 +1,130 @@
+---
+title: Upgrading Calico 
+canonical_url: 'https://docs.projectcalico.org/master/getting-started/openstack/upgrade/upgrade'
+---
+
+## Calico package update
+
+This part of the upgrade procedure varies slightly according to which OS you are using.  
+
+- [Upgrading an Openstack cluster based on Centos](#upgrading-an-openstack-cluster-based-on-centos)
+
+- [Upgrading an Openstack cluster based on Ubuntu](#upgrading-an-openstack-cluster-based-on-ubuntu)
+
+> **Important**: Do not use older versions of `calicoctl` after the upgrade.
+> This may result in unexpected behavior and data.
+{: .alert .alert-danger}
+
+
+## Upgrading an Openstack cluster based on Centos
+   
+1. On all nodes, change the location of the Calico packages to point to the 3.x repo:
+
+   ```
+   sudo sed -i 's/calico-2.6/calico-3.1/g' /etc/yum.repos.d/calico.repo 
+   ```
+
+1. Upgrade the compute software
+   
+   1. On all compute nodes, update packages:
+      ```
+      sudo yum update
+      ```
+      We recommend upgrading the whole distribution as shown here. In case you prefer to upgrade particular packages only, those needed for a Calico compute node are the following.
+      ```
+      calico-common
+      calico-compute
+      calico-dhcp-agent
+      calico-felix
+      dnsmasq
+      networking-calico
+      openstack-neutron
+      openstack-nova-api
+      openstack-nova-compute
+      ```
+      
+   1. Use the following command on the compute nodes to confirm that Felix has upgraded to v3.1.x.
+      ```
+      calico-felix --version
+      ```
+   
+      It should return `v3.1.x`.
+   
+   1. On all compute nodes, add the following line to the end of `/etc/calico/felix.cfg`:
+      ```
+      DatastoreType = etcdv3
+      ```
+   
+1. Upgrade the Control software:
+   1. On all control nodes, update packages:
```

Comment:
Do these nested numberings render as intended?

Reply:
They do in pycharm :)  I need `make serve` to work to see if it works for real.